src/components/calculator.js

```diff
@@ -1,15 +1,39 @@
-/* eslint-disable react/button-has-type */
 import { PropTypes } from 'prop-types';
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import calculate from '../logic/calculate';
 
-// let counter = 0;
 const ans = document.querySelector('.answer');
 
 const Calculator = () => {
   const [totalVal, setTotalVal] = useState(null);
   const [nextVal, setNextVal] = useState(null);
   const [operationVal, setOperationVal] = useState(null);
+  const [data, setData] = useState([]);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      try {
+        const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=learning', {
+          headers: {
+            'X-Api-Key': 'baoB69sSSBvXDFb2gkXx6w==5nrM0rh00PkzY7G0',
+          },
+        });
+        const json = await res.json();
+        setData(json);
+      } catch (error) {
+        setHasError(true);
+      }
+      setIsLoading(false);
+    };
+    fetchData();
+  }, [setData, setIsLoading]);
+
+  if (hasError) return <div>Something went wrong!</div>;
+
+  if (isLoading) return <div>Loading...</div>;
```

Comment:
- [ ] Excellent work for fetching the quotes from external API correctly. However, You must create a separate `Quotes` component and put everything related to quotes there instead of adding everything in just one component. By creating a separate component your codebase will be clean and professional and even follow the best practices in react of creating a reusable component. Kindly fix that.

Reply:
Fixed the problem as per instructions. Thanks for the valuable suggestion. I really appreciate it.